runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/external/PythonExecutorLifecycleManagerImpl.java

```diff
@@ -0,0 +1,215 @@
+import org.apache.logging.log4j.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static io.cloudslang.runtime.api.python.enums.PythonStrategy.PYTHON_EXECUTOR;
+import static io.cloudslang.runtime.api.python.enums.PythonStrategy.getPythonStrategy;
+import static java.io.File.separator;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static org.jboss.resteasy.util.HttpHeaderNames.AUTHORIZATION;
+import static org.jboss.resteasy.util.HttpHeaderNames.CONTENT_TYPE;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON;
+
+@Service("pythonExecutorLifecycleManager")
+public class PythonExecutorLifecycleManagerImpl implements PythonExecutorLifecycleManager {
+
+    private static final Logger logger = LogManager.getLogger(PythonExecutorLifecycleManagerImpl.class);
+    private static final PythonStrategy PYTHON_EVALUATOR = getPythonStrategy(System.getProperty("python.expressionsEval"), PYTHON_EXECUTOR);
+    private static final String EXTERNAL_PYTHON_EXECUTOR_STOP_PATH = "/rest/v1/stop";
+    private static final String EXTERNAL_PYTHON_EXECUTOR_HEALTH_PATH = "/rest/v1/health";
+    private static String PYTHON_EXECUTOR_PORT;
+    private static String EXTERNAL_PYTHON_EXECUTOR_URL;
+    private static String ENCODED_AUTH;
+    private static String PYTHON_EXECUTOR_BIN_PATH;
+    private static ResteasyClient restEasyClient;
+    private static Process pythonExecutorProcess;
+
+    @Autowired
+    private StatefulRestEasyClientsHolder statefulRestEasyClientsHolder;
+
+    @Autowired
+    @Qualifier("pythonExecutorConfigurationDataService")
+    private PythonExecutorConfigurationDataService pythonExecutorConfigurationDataService;
+
+    @PostConstruct
+    void initPythonExecutorDetails() {
+        PythonExecutorDetails pythonExecutorDetails = pythonExecutorConfigurationDataService.getPythonExecutorConfiguration();
+        PYTHON_EXECUTOR_PORT = pythonExecutorDetails.getPort();
+        EXTERNAL_PYTHON_EXECUTOR_URL = pythonExecutorDetails.getUrl();
+        ENCODED_AUTH = pythonExecutorDetails.getLifecycleEncodedAuth();
+        PYTHON_EXECUTOR_BIN_PATH = pythonExecutorDetails.getPythonExecutorPath() + separator + "bin";
+        restEasyClient = statefulRestEasyClientsHolder.getRestEasyClient();
+        if (PYTHON_EVALUATOR.equals(PYTHON_EXECUTOR)) {
+            doStartPythonExecutor();
+        }
+    }
+
+    private void doStartPythonExecutor() {
+        start();
+    }
+
+    @Override
+    public void start() {
+        logger.info("A request to start the Python Executor was sent");
```

Comment:
Have this code in doStartPythonExecutor. A public method should call the private one, not the other way around.